nbgitpuller/plugins/README.md

```diff
@@ -0,0 +1,23 @@
+# [nbgitpuller download plugins](https://github.com/jupyterhub/nbgitpuller)
```

Comment:
Could you write a bit more generally about plugin system in this file? This would be documentation meant for maintainers and someone that considers if a plugin could be relevant to develop in order to augment the functionality of nbgitpuller in some way.

I'm thinking about notes similar to this.

- nbgitpuller currently allows its plugins to do ... specifically, but not for example to do ...
- nbgitpuller plugins can register themselves automatically by declaring ... in setup.py
- all nbgitpuller plugins must implement ...
- nbgitpuller ships with the following plugins...

Reply:
I think I got all this. good deal